spec/board_spec.rb

```diff
@@ -30,15 +30,49 @@
   end
 
   describe "#game_over?" do
-    context "when the board has an available space" do
-      it "returns false" do
-        expect(board.game_over?).to eq(false)
+    def fill_board
+      [1, 2, 3, 4, 6, 5, 7, 9, 8].each { |pos| board.place_token(pos) }
+    end
+
+    def set_diagonal_winner
+      (1..7).each { |pos| board.place_token(pos) }
+    end
+
+    def set_column_winner
+      [1, 2, 4, 3, 7].each { |pos| board.place_token(pos) }
+    end
+
+    def set_row_winner
+      [1, 4, 2, 7, 3].each { |pos| board.place_token(pos) }
+    end
+
+    context "with a diagonal winner" do
+      it "returns true" do
+        set_diagonal_winner
+
+        expect(board.game_over?).to eq(true)
+      end
+    end
+
+    context "with a column winner" do
+      it "returns true" do
+        set_column_winner
+
+        expect(board.game_over?).to eq(true)
+      end
+    end
+
+    context "with a row winner" do
+      it "returns true" do
+        set_row_winner
+
+        expect(board.game_over?).to eq(true)
       end
     end
 
-    context "when the board does not have an available space" do
+    context "with a tie" do
       it "returns true" do
-        (1..9).each { |pos| board.place_token(pos) }
+        fill_board
```

Comment:
I wonder if we want to be more explicit and rename this method to something that calls out that it is a draw. Thoughts?

Reply:
I actually did on the subsequent pull request.

I hard coded the rows, columns, and diagonals instead of using an algorithm to determine the rows, columns, and diagonals.  The algorithm would be beneficial if we had a Tic-Tac-Toe board with a different dimension, but Damon and I thought hard coding met the current requirements and was easy to implement.